Add AT command parser example to docs (#122)
ARMmbed/mbed-os-examples-docs_only: APIs_Platform/AT_CmdParser/README.md

## ATCmdParser Example ##

> This is a example showing ATCmdParser usage with ESP8266 WiFi module.

Reviewer: ```suggestion
Example application showing usage of the `ATCmdParser` class with the ESP8266 WiFi module.
```

> Example application showing usage of the `ATCmdParser` class with the ESP8266 WiFi module.

Reviewer 2: I did not think we have README files with code snippets.

Author: @evedon the README is optional, although, most of the snippets include one. I've removed extraneous information and placed some of it in the docs section describing the snippet.

The program retrieves the FW version of the ESP8266 module connected over UART using AT commands. 

##  Getting started

> 1. Import the example
> 2. Compile and generate binary   
> 5. Open a serial console session with the target platform using the following parameters:
>     * **Baud rate:** 9600
>     * **Data bits:** 8
>     * **Stop bits:** 1
>     * **Parity:** None
>  6. Flash the target with the compiled binary. 
>  7. The serial console should display a similar output to below, indicating a successful AT communication:

Reviewer: ```suggestion
1. Import the example
1. Compile and generate binary   
1. Open a serial console session with the target platform using the following parameters:
    * **Baud rate:** 9600
    * **Data bits:** 8
    * **Stop bits:** 1
    * **Parity:** None
 1. Flash the target with the compiled binary. 
 1. The serial console should display a similar output to below, indicating a successful AT communication:
```

> 1. The serial console should display a similar output to below, indicating a successful AT communication:

Reviewer 2: This is not an example so I was not expecting a "getting started" section. We tend to use a template in the Mbed OS examples and recommend using "mbed compile -m <TARGET> -t <TOOLCHAIN> --flash --sterm"
